### PR TITLE
fix(core): Fix validation of empty list tool message content

### LINKED ISSUE #33758: Empty list as valid tool message content
### Checked other resources

- [x] This is a bug, not a usage question.
- [x] I added a clear and descriptive title that summarizes this issue.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [x] I am sure that this is a bug in LangChain rather than my code.
- [x] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).
- [x] This is not related to the langchain-community package.
- [x] I read what a minimal reproducible example is (https://stackoverflow.com/help/minimal-reproducible-example).
- [x] I posted a self-contained, minimal, reproducible example. A maintainer can copy it and run it AS IS.

### Example Code

```python
# the bug code
def _is_message_content_type(obj: Any) -> bool:
    """Check if object is valid message content format.

    Validates content for OpenAI or Anthropic format tool messages.

    Args:
        obj: The object to check.

    Returns:
        `True` if the object is valid message content, `False` otherwise.
    """
    return isinstance(obj, str) or (
        isinstance(obj, list) and all(_is_message_content_block(e) for e in obj)
    )


def _is_message_content_block(obj: Any) -> bool:
    """Check if object is a valid message content block.

    Validates content blocks for OpenAI or Anthropic format.

    Args:
        obj: The object to check.

    Returns:
        `True` if the object is a valid content block, `False` otherwise.
    """
    if isinstance(obj, str):
        return True
    if isinstance(obj, dict):
        return obj.get("type", None) in TOOL_MESSAGE_BLOCK_TYPES
    return False

# should be Flase rather than True, True will result in the corresponding content in the request not being a string; The request will result in an error.
_is_message_content_type([])
```

request body like:
Image

tool`s content is not a str

then error message is 402(DeepSeek Platform):
Failed to deserialize the JSON body into the target type: messages[3]: invalid type: sequence, expected a string at line 1 column 816

### Error Message and Stack Trace (if applicable)

Failed to deserialize the JSON body into the target type: messages[3]: invalid type: sequence, expected a string at line 1 column 816

### Description

I'm building my own agent using the LangChain framework, which incorporates the Tools capability. During a remote large model call that combined Tools results, I encountered a 402 error. Upon investigation, I discovered this occurs when the `obj` type is a list but the list is empty. I first optimized my own code to avoid this issue, but I believe the `langchain_core` module should ensure robustness for this scenario.

### System Info

System Information
------------------
> OS:  Darwin
> OS Version:  Darwin Kernel Version 25.0.0: Wed Sep 17 21:41:50 PDT 2025; root:xnu-12377.1.9~141/RELEASE_ARM64_T6030
> Python Version:  3.14.0 (v3.14.0:ebf955df7a8, Oct  7 2025, 08:20:14) [Clang 16.0.0 (clang-1600.0.26.6)]

Package Information
-------------------
> langchain_core: 1.0.1
> langchain: 1.0.2
> langchain_community: 0.4.1
> langsmith: 0.4.38
> langchain_classic: 1.0.0
> langchain_openai: 1.0.1
> langchain_tavily: 0.2.12
> langchain_text_splitters: 1.0.0
> langgraph_sdk: 0.2.9

Optional packages not installed
-------------------------------
> langserve

Other Dependencies
------------------
> aiohttp: 3.13.2
> async-timeout: Installed. No version info available.
> claude-agent-sdk: Installed. No version info available.
> dataclasses-json: 0.6.7
> httpx: 0.28.1
> httpx-sse: 0.4.3
> jsonpatch: 1.33
> langchain-anthropic: Installed. No version info available.
> langchain-aws: Installed. No version info available.
> langchain-deepseek: Installed. No version info available.
> langchain-fireworks: Installed. No version info available.
> langchain-google-genai: Installed. No version info available.
> langchain-google-vertexai: Installed. No version info available.
> langchain-groq: Installed. No version info available.
> langchain-huggingface: Installed. No version info available.
> langchain-mistralai: Installed. No version info available.
> langchain-ollama: Installed. No version info available.
> langchain-perplexity: Installed. No version info available.
> langchain-together: Installed. No version info available.
> langchain-xai: Installed. No version info available.
> langgraph: 1.0.1
> langsmith-pyo3: Installed. No version info available.
> numpy: 2.3.4
> openai: 2.6.1
> openai-agents: Installed. No version info available.
> opentelemetry-api: Installed. No version info available.
> opentelemetry-exporter-otlp-proto-http: Installed. No version info available.
> opentelemetry-sdk: Installed. No version info available.
> orjson: 3.11.4
> packaging: 25.0
> pydantic: 2.12.3
> pydantic-settings: 2.11.0
> pytest: Installed. No version info available.
> pyyaml: 6.0.3
> PyYAML: 6.0.3
> requests: 2.32.5
> requests-toolbelt: 1.0.0
> rich: Installed. No version info available.
> SQLAlchemy: 2.0.44
> sqlalchemy: 2.0.44
> tenacity: 9.1.2
> tiktoken: 0.12.0
> typing-extensions: 4.15.0
> vcrpy: Installed. No version info available.
> zstandard: 0.25.0

### PR DESCRIPTION
**Pull request by BushGommern**
Fixes #33758

### Summary

This PR fixes an edge case in tool message content validation where an empty list (`[]`) was previously considered valid message content.

While non-empty lists of valid content blocks are supported, an empty list results in an invalid request payload for several providers (e.g. DeepSeek), which expect tool message content to be a string. This could lead to request deserialization errors at runtime.

### What was changed

- Updated `_is_message_content_type` to treat empty lists as invalid message content
- Added a unit test to cover this edge case and prevent regressions

### Why this is needed

Previously, `_is_message_content_type([])` returned `True`, allowing empty list content to pass validation. This led to malformed JSON request bodies and downstream provider errors such as:invalid type: sequence, expected a string

Rejecting empty lists at validation time makes message handling more robust and avoids generating invalid requests.

### Tests

- Added a unit test case to ensure empty list content is correctly rejected
- All existing tests continue to pass

### CODE REVIEW

File: libs/core/langchain_core/tools/base.py [hunk 1]
```diff
@@ -1271,6 +1271,8 @@ def _is_message_content_type(obj: Any) -> bool:
     Returns:
         `True` if the object is valid message content, `False` otherwise.
     """
+    if isinstance(obj, list) and not obj:  # Check if it's an empty list
```
Comment (mdrxy) [inline libs/core/langchain_core/tools/base.py]:
```suggestion
    if isinstance(obj, list) and not obj:
```

File: libs/core/langchain_core/tools/base.py [hunk 2]
```diff
@@ -1271,6 +1271,8 @@ def _is_message_content_type(obj: Any) -> bool:
...
+        return False  # Empty lists should be considered invalid
```
Comment (mdrxy) [inline libs/core/langchain_core/tools/base.py]:
```suggestion
        return False
```

**Comment by mdrxy** [comment]
https://github.com/langchain-ai/langchain/issues/33758#issuecomment-3693791464
